common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -17,13 +21,276 @@
+                            struct nlattr *tb_freq[NL80211_FREQUENCY_ATTR_MAX + 1];
+                            sChannelInfo channel{};
+
+                            if (nla_parse(tb_freq, NL80211_FREQUENCY_ATTR_MAX,
+                                          static_cast<nlattr *>(nla_data(nl_freq)),
+                                          nla_len(nl_freq), freq_policy)) {
+                                LOG(ERROR)
+                                    << "Failed to parse frequency " << nl_freq->nla_type + 1 << "!";
+                                return false;
+                            }
+
+                            if (!tb_freq[NL80211_FREQUENCY_ATTR_FREQ] ||
+                                tb_freq[NL80211_FREQUENCY_ATTR_DISABLED] ||
+                                (tb_freq[NL80211_FREQUENCY_ATTR_NO_IR] &&
+                                 tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) ||
+                                tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) {
+                                continue;
+                            }
+
+                            if (tb_band[NL80211_FREQUENCY_ATTR_FREQ]) {
+                                uint32_t freq   = nla_get_u32(tb_freq[NL80211_FREQUENCY_ATTR_FREQ]);
+                                channel.channel = ieee80211_frequency_to_channel(freq);
+                            } else {
+                                LOG(DEBUG) << "NL80211_FREQUENCY_ATTR_FREQ attribute is missing";
+                            }
+
+                            if (tb_freq[NL80211_FREQUENCY_ATTR_MAX_TX_POWER]) {
+                                channel.tx_pow_dBm =
+                                    0.01 *
+                                    nla_get_u32(tb_freq[NL80211_FREQUENCY_ATTR_MAX_TX_POWER]);
+                            } else {
+                                LOG(DEBUG)
+                                    << "NL80211_FREQUENCY_ATTR_MAX_TX_POWER attribute is missing";
+                            }
+
+                            if (tb_freq[NL80211_FREQUENCY_ATTR_RADAR]) {
+                                channel.is_dfs = true;
+                            }
+
+                            if (!tb_freq[NL80211_FREQUENCY_ATTR_NO_20MHZ]) {
+                                channel.bandwidth_MHz = 20;
+                            }
+                            if (ctx.width_40 && (!tb_freq[NL80211_FREQUENCY_ATTR_NO_HT40_MINUS] ||
+                                                 !tb_freq[NL80211_FREQUENCY_ATTR_NO_HT40_PLUS])) {
+                                channel.bandwidth_MHz = 40;
+                            }
+                            if (ctx.width_80 && !tb_freq[NL80211_FREQUENCY_ATTR_NO_80MHZ]) {
+                                channel.bandwidth_MHz = 80;
+                            }
+                            if (ctx.width_160 && !tb_freq[NL80211_FREQUENCY_ATTR_NO_160MHZ]) {
+                                channel.bandwidth_MHz = 160;
+                            }
```

Comment:
I'm not sure how this will be used, but I believe we in the end will need a sWifiChannel for each separate bandwidth at which we can operate. At least, that's the assumption when constructing the EasyMesh messages (AP Radio Basic Capabilities). @kantera800 @YuriyMasechko @vitalybu please comment.

In practice, of course, the maximum bandwidth is sufficient since we can (probably) always operate at a lower bandwidth.

Reply:
Supported channels in this new implementation are obtained the same way as in previous implementation (which I guess will soon be replaced by this one).
I've also seen that nl80211 command returns a nested array of operating bands in `NL80211_ATTR_WIPHY_BANDS`. However, previous implementation considers just one band, the first one, and ignores the rest

Reply:
Modified as it was agreed during meeting: each channel includes a vector of supported_bandwidths.
I've also added a method to get maximum bandwidth supported